src/js/actions/export.js

```diff
@@ -0,0 +1,365 @@
+
+    /**
+     * Set the format of the asset specified by the given index
+     *
+     * @param {Document} document
+     * @param {Layer} layer
+     * @param {number} assetIndex index of this asset within the layer's list
+     * @param {string} format (example: jpg)
+     * @return {Promise}
+     */
+    var updateLayerAssetFormat = function (document, layer, assetIndex, format) {
+        return this.transfer(updateLayerExportAsset, document, layer, assetIndex, { format: format });
+    };
+    updateLayerAssetFormat.reads = [];
+    updateLayerAssetFormat.writes = [];
+    updateLayerAssetFormat.transfers = [updateLayerExportAsset];
+
+    /**
+     * Update the status of all assets that are being requested
+     *
+     * @param {Document} document [description]
+     */
+    var setAllAssetsRequested = function (document) {
+        var layerIDs = document.layers.all.reduce(function (IDs, layer) {
+            if (layer.exportEnabled) {
+                IDs.push(layer.id);
+            }
+            return IDs;
+        }, []);
+        return this.dispatchAsync(events.export.SET_AS_REQUESTED, { documentID: document.id, layerIDs: layerIDs });
+    };
+    setAllAssetsRequested.reads = [];
+    setAllAssetsRequested.writes = [locks.JS_DOC];
+
+    /**
+     * Delete the Export Asset configuration specified by the given index
+     *
+     * @param {Document} document
+     * @param {Layer} layer
+     * @param {number} assetIndex index of this asset within the layer's list
+     * @return {Promise}
+     */
+    var deleteLayerExportAsset = function (document, layer, assetIndex) {
+        var documentID = document.id,
+            layerID = layer.id,
+            payload = {
+                documentID: documentID,
+                layerID: layerID,
+                assetIndex: assetIndex
+            };
+
+        return this.dispatchAsync(events.export.DELETE_LAYER_ASSET, payload).bind(this).then(function () {
+            return _syncLayerExportMetadata.call(this, documentID, layer.id);
+        });
+    };
+    deleteLayerExportAsset.reads = [locks.JS_DOC];
+    deleteLayerExportAsset.writes = [locks.JS_DOC, locks.PS_DOC];
+
+    /**
+     * Sets the exportEnabled flag for a given layer or layers
+     * @private
+     * @param {Document} document Owner document
+     * @param {Layer|Immutable.List.<Layer>} layers Either a Layer reference or array of Layers
+     * @param {boolean=} exportEnabled
+     * @returns {Promise}
+     */
+    var setLayerExportEnabled = function (document, layers, exportEnabled) {
+        var layerIDs = Immutable.List.isList(layers) ?
+                collection.pluck(layers, "id").toArray() :
+                [layers.id];
+
+        var payload = {
+                documentID: document.id,
+                layerIDs: layerIDs,
+                exportEnabled: exportEnabled
+            };
+
+        this.dispatch(events.document.LAYER_EXPORT_ENABLED_CHANGED, payload);
```

Comment:
In this case I am relying on the synchronicity of the dispatch.  I actually originally had `dispatchAsync(...).then()` ... but I felt it read better this way.  I've added a comment here, and added a further explanation in `_syncLayerExportMetadata` which I had meant to do anyway.  Let's discuss further if you're still not buying it :money_with_wings: